src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java

```diff
@@ -2904,4 +2909,52 @@ private void getAdditionalRequirementsForHearingReqForSdo(StandardDirectionOrder
             );
         }
     }
+
+    public List<String> validateRespondentLipAndOtherPersonAddress(CaseData caseData) {
+        List<String> errorList = new ArrayList<>();
+        if (null != caseData.getManageOrders().getRecipientsOptions()
+            && No.equals(caseData.getManageOrders().getServeToRespondentOptions())) {
+            List<String> selectedRespondentIds = caseData.getManageOrders().getRecipientsOptions().getValue()
+                .stream().map(DynamicMultiselectListElement::getCode).toList();
+            validateAddressForParty(caseData.getRespondents(), selectedRespondentIds, errorList, true);
+
+        }
+        if (null != caseData.getManageOrders().getOtherParties() && null != caseData.getOtherPartyInTheCaseRevised()) {
+            List<String> selectedOtherPartyIds = caseData.getManageOrders().getOtherParties().getValue()
+                .stream().map(DynamicMultiselectListElement::getCode).toList();
+            validateAddressForParty(caseData.getOtherPartyInTheCaseRevised(), selectedOtherPartyIds, errorList, false);
+        }
+        return errorList;
+    }
+
+    private void validateAddressForParty(List<Element<PartyDetails>> partyDetails,
```

Comment:
This method needs to be refactored.

Reply:
@shashisk97 as discussed renamed this method to validate the address and return error.